contracts/oracle/CollateralizationOracleWrapper.sol

```diff
@@ -0,0 +1,272 @@
+    /// @notice update reading of the CollateralizationOracle
+    function update() external override whenNotPaused {
+        // fetch a fresh round of information
+        (
+            uint256 _protocolControlledValue,
+            uint256 _userCirculatingFei,
+            int256 _protocolEquity,
+            bool _validityStatus
+        ) = ICollateralizationOracle(collateralizationOracle).pcvStats();
+
+        // only update if valid
+        require(_validityStatus, "CollateralizationMemoizer: CollateralizationOracle is invalid");
+
+        // set cache variables
+        cachedProtocolControlledValue = _protocolControlledValue;
+        cachedUserCirculatingFei = _userCirculatingFei;
+        cachedProtocolEquity = _protocolEquity;
+
+        // reset time
+        _initTimed();
+
+        // emit event
+        emit CachedValueUpdate(
+            msg.sender,
+            cachedProtocolControlledValue,
+            cachedUserCirculatingFei,
+            cachedProtocolEquity
+        );
+    }
+
+    // @notice returns true if the cached values are outdated.
+    function isOutdated() public override view returns (bool outdated) {
+        // check if cached value is fresh
+        return isTimeEnded();
+    }
+
+    /// @notice Get the current collateralization ratio of the protocol, from cache.
+    /// @return collateralRatio the current collateral ratio of the protocol.
+    /// @return validityStatus the current oracle validity status (false if any
+    ///         of the oracles for tokens held in the PCV are invalid, or if
+    ///         this contract is paused).
+    function read() external view override returns (Decimal.D256 memory collateralRatio, bool validityStatus) {
+        collateralRatio = Decimal.ratio(cachedProtocolControlledValue, cachedUserCirculatingFei);
+        validityStatus = !paused() && !isOutdated();
+    }
+
+    // ----------- Wrapper-specific methods ------------------------------------
+    // @notice returns true if the cached values are obsolete, i.e. the actual CR
+    //         readings deviated from cached value by more than a threshold.
+    //         This function is intended to be called off-chain by keepers.
+    //         If executed on-chain, it consumes more gas than an actual update()
+    //         call _and_ does not persist the read values in the cached state.
+    function isExceededDeviationThreshold() public view returns (bool obsolete) {
+        (
+            uint256 _protocolControlledValue,
+            uint256 _userCirculatingFei,
+            , // int256 _protocolEquity,
+            bool _validityStatus
+        ) = ICollateralizationOracle(collateralizationOracle).pcvStats();
+
+        require(_validityStatus, "CollateralizationMemoizer: CollateralizationOracle reading is invalid");
+
+        Decimal.D256 memory _thresholdLow = Decimal.from(10_000 - deviationThresholdBasisPoints).div(10_000);
```

Comment:
I still think a helper method would be cleaner but this looks fine so I won't die on that hill